src/vs/workbench/contrib/chat/browser/chatInputPart.ts

```diff
@@ -571,6 +577,57 @@ export class ChatInputPart extends Disposable implements IHistoryNavigationWidge
 				});
 				widget.ariaLabel = ariaLabel;
 				widget.tabIndex = 0;
+			} else if (attachment.isImage) {
+				const blob = new Blob([attachment.value as Uint8Array], { type: 'image/png' });
+				const url = URL.createObjectURL(blob);
+
+				const img = document.createElement('img');
+				img.classList.add('chat-attached-context-image');
+				img.src = url;
+				img.alt = '';
+
+				// Pill with tiny image
+				const pill = document.createElement('div');
+				pill.classList.add('chat-attached-context-pill');
+				pill.tabIndex = -1;
+
+				const pillImg = document.createElement('img');
+				pillImg.src = url;
+				pillImg.alt = '';
+				pillImg.classList.add('chat-attached-context-pill-image');
+
+				pill.appendChild(pillImg);
+
+				// Custom label
+				const textLabel = document.createElement('span');
+				textLabel.textContent = attachment.name;
+				textLabel.classList.add('chat-attached-context-custom-text');
+
+
+				widget.style.position = 'relative';
+				widget.appendChild(pill);
+				widget.appendChild(textLabel);
+
+				const ariaLabel = localize('chat.imageAttachment', "Attached image, {0}", attachment.name);
+
+				widget.ariaLabel = ariaLabel;
+				widget.tabIndex = 0;
+
+				// Hover
+				const hoverElement = document.createElement('div');
+				hoverElement.appendChild(img);
+				hoverElement.classList.add('chat-attached-context-hover');
+				hoverElement.setAttribute('aria-label', ariaLabel);
+
+				this._register(this.hoverService.setupManagedHover(getDefaultHoverDelegate('mouse'), widget, hoverElement));
+
+				// No delay for keyboard
+				this._register(dom.addDisposableListener(widget, 'keydown', (event) => {
```

Comment:
What about having the hover show as soon as the user focuses on it? This is what we normally do. I doubt users will figure out that they need to focus the item and press `Enter` or `Space`.

Reply:
the `setUpManagedHover` does this for us (it will show hover after a slight delay). I was wondering if we should show immediately vs. having delay, but most hovers it seems are shown after a delay.

Reply:
We don't recommend removing the delay at the start as it can be pretty annoying in some cases. However, what I recommend is to create an instant hover delegate (`createInstantHoverDelegate()` to create, but needs to be disposed) outside this loop and use the same instance for all managed hovers. This would allow you to tab through the attachments and have the hovers show instantly after having to wait the delay the first time.